proto/buf/alpha/registry/v1alpha1/image.proto

```diff
@@ -34,3 +37,53 @@ message GetImageRequest {
 message GetImageResponse {
   buf.alpha.image.v1.Image image = 1;
 }
+
+enum PartialImageLevel {
+  PARTIAL_IMAGE_LEVEL_UNSPECIFIED = 0;
+
+  // Returns complete files in which the requested types are contained.
+  PARTIAL_IMAGE_LEVEL_FILES = 1;
+
+  // Returns trimmed files in which the requested types are contained. In a
+  // trimmed file, all declarations that are not embedded (directly or
+  // indirectly) in requested types are removed.
+  PARTIAL_IMAGE_LEVEL_TYPES = 2;
+}
+
+message GetPartialImageRequest {
+  string owner = 1;
+  string repository = 2;
+  string reference = 3;
+
+  // If the image should exclude depdendencies. When set to true, a incomplete
+  // image without descriptors for dependencies of the requested types is
+  // returned.
+  bool exclude_dependencies = 4;
+
+  // Determines if the patterns specified in {messages,enums,services} should
+  // consider types from imported modules.
+  bool match_dependencies = 5;
+  
+  // Type names or patterns describing which messages should be included in the
+  // partial image, e.g. 
+  // - "pkg.foo.Bar" for a message named 'Bar' in 'pkg.foo'
+  // - "pkg.foo.*" for all messages in package 'pkg.foo'
+  // - "pkg.foo.*.**" for all messages under 'pkg.foo' and subpackages
+  // - "*.**" for all messages in all packages.
+  repeated string messages = 6;
+  // Type names or patterns describing which enums should be included in the
+  // partial image. For examples see `messages`.
+  repeated string enums = 7;
+  // Type names or patterns describing which services should be included in the
+  // partial image. For examples see `messages`.
+  repeated string services = 8;
```

Comment:
Talking to @bufdev, should these all be collapsed into a single `types`? If we implement this in bufmodule in general and allow partial images on types from the cli too, a single type may be a better flag experience.

I had originally separated them so it would be possible to get an image containing only the service definitions (without the messages+imports for their request/response etc) as this satisfies the GetServices needs of what is now in the RequestBuilder service. There may be other options there, like using something like fieldmask.

Reply:
I agree with consolidating in a single `types` for the symmetry you calling out. Having something like `buf build --type pet.v1.Pet --type payment.v2.Order` would be great.

I wanted to open an issue to discuss a tree-shaking solution like `buf build --only pet.v1.Pet`, but `--type` is just as clear, if not more so. It's also symmetrical with other potential commands like `buf decode`.

Reply:
Yea, I prefer dropping the leading `.` prefix since it's generally just simpler and cleaner to read. This is also what the [protoreflect.FullName](https://github.com/protocolbuffers/protobuf-go/blob/e5db2960ed1380681b571cdf4648230beefaf58b/reflect/protoreflect/proto.go#L438) representation does.

Wildcards sound cool, but I doubt we'll need to support them at first.

Reply:
we drop the `.` for full names across the board